modules/RTC/LocalSdpMunger.js

```diff
@@ -0,0 +1,288 @@
+/* global __filename */
+
+import { getLogger } from 'jitsi-meet-logger';
+import * as MediaType from '../../service/RTC/MediaType';
+import { SdpTransformWrap } from '../xmpp/SdpTransformUtil';
+
+const logger = getLogger(__filename);
+
+/**
+ * Fakes local SDP, so that it will reflect current local tracks status inside
+ * of the {@link TraceablePeerConnection} and make operations like
+ * attach/detach and video mute/unmute local operations. That means it prevents
+ * from SSRC updates being sent to Jicofo/remote peer, so that there is no
+ * sRD/sLD cycle on the remote side.
+ *
+ * FIXME audio SSRC is not consistent, between attach and detach
+ */
+export default class LocalSdpMunger {
+
+    /**
+     * Creates new <tt>LocalSdpMunger</tt> instance.
+     *
+     * @param {TraceablePeerConnection} tpc
+     */
+    constructor(tpc) {
+        this.tpc = tpc;
+    }
+
+    /**
+     * Makes sure local SDP for audio media reflects current local tracks status
+     * of the parent TPC.
+     * @param {SdpTransformWrap} transformer the transformer instance which will
+     * be used to process the SDP.
+     * @return {boolean} <tt>true</tt> if any modification were made.
+     * @private
+     */
+    _describeLocalAudioTracks(transformer) {
```

Comment:
I will try to improve, but any suggestions would be welcome. I'm not sure which details I'm missing here, but that's because I know what it does.